src/webgpu/api/validation/encoding/queries/general.spec.ts

```diff
@@ -89,10 +89,8 @@ Tests that write timestamp to all types of query set on all possible encoders:
       .expand('queryIndex', p => (p.type === 'timestamp' ? [0, 2] : [0]))
   )
   .beforeAllSubcases(t => {
-    const { type } = t.params;
-    if (type) {
-      t.selectDeviceForQueryTypeOrSkipTestCase(type);
-    }
+    // writeTimestamp is only available for devices that enable the 'timestamp-query' feature.
+    t.selectDeviceForQueryTypeOrSkipTestCase('timestamp');
```

Comment:
why not `t.params.type`?

Reply:
I'm silly - that's what it already was; why had it not properly skipped?

Reply:
Because then `writeTimestamp()` will be called and throw an exception when the query type is `occlusion` because `timestamp-query` wasn't enabled. We can wrap the `writeTimestamp()` in a `t.shouldThrow()` but then you've removed the entire thing you're supposed to be testing and replaced it with a test for "Does this function throw if the feature isn't enabled?" (Which is already being tested in `webgpu:api,validation,capability_checks,features,query_types:writeTimestamp:*`)

Reply:
OH it's because it is *always* doing writeTimestamp, with potentially non-timestamp queries.

Sounds like we need to skip if either t.params.type OR 'timestamp' is unsupported?

Reply:
Ah, good point! We might have future query types that require their own independent features. I'll make that adjustment.

Reply:
Okay, updated. Should be more future-proof now. Thanks!